File: Chapter5/5.8/reverse/java/reverse.java
import java.util.Scanner;

public class reverse {
    public static int reverseDigit(int x){
	int sign = 1;
	if (x<0) sign = -1;
	x *= sign;
	
	int result=0;
	while (x>0) {
	    result *= 10;
	    result += x%10;
	    x /= 10;
	}

	return result*sign;
    }

    public static void main(String[] args){
	Scanner in = new Scanner(System.in);
	int x=in.nextInt();
	System.out.println("reverse:"+reverseDigit(x));
    }
}

File: Chapter5/5.8/reverse/cpp/reverse.cpp
#include <iostream>

using namespace std;

int reverse(int x){
  int sign = 1;
  if (x<0) sign=-1;
  int result = 0;
  while (x>0) {
    result *= 10;
    result += x%10;
    x /= 10;
  }
  return result*sign;
}

int main(){
  int x;
  cin>>x;
  cout<<"reverse:"<<reverse(x)<<endl;
}
